hayeon/programmers-1/문자열-나누기.js

```diff
@@ -0,0 +1,49 @@
+function solution(s) {
+  let result = 0;
+
+  for (let i = 0; i < s.length; i++) {
+    let sameStr = 1;
+    let diffStr = 0;
+
+    for (let j = i + 1; j < s.length; j++) {
+      if (s[i] === s[j]) {
+        sameStr++;
+      } else {
+        diffStr++;
+      }
+
+      if (sameStr === diffStr) {
+        result++;
+        i = j;
+        break;
+      }
+
+      if (j + 1 === s.length) {
+        i = j;
+      }
+    }
+
+    if (sameStr !== diffStr && i + 1 === s.length) {
+      result++;
+    }
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(s) {
  let result = 0;

  for (let i = 0; i < s.length; i++) {
    let sameCount = 1;
    let diffCount = 0;

    for (let j = i + 1; j < s.length; j++) {
      if (s[i] === s[j]) {
        sameCount++;
      } else {
        diffCount++;
      }

      if (sameCount === diffCount) {
        result++;
        i = j;
        break;
      }

      if (j + 1 === s.length) {
        i = j;
      }
    }

    if (sameCount !== diffCount && i + 1 === s.length) {
      result++;
    }
  }
  return result;
}
```
- 풀이잘봤습니다! 🙇‍♂️
- 아래 주석으로 말씀주신대로 카운트하는 변수를 활용하셨는데 변수명이 아쉽네요 ~ Str을 문자열을 의미하기때문에 카운팅 목적인 변수에는 적절하지 않아보입니다. Count 변수명을 붙여주세요!

Reply:
오,,, 그렇네요 그런부분을 생각하지 못했다니.... 감사합니다!!!!!😮